spec/mt940_parsing_spec.rb

```diff
@@ -95,4 +95,115 @@
       expect(subject[0].transactions.count).to eql(1)
     end
   end
+
+  describe "MT940 abnamro" do
+    it "splits the file into on statement with the headers included" do
+      expected_data = <<~MT940.chomp
+        ABNANL2A
+        940
+        ABNANL2A
+        :20:ABN AMRO BANK NV
+        :25:517852257
+        :28:19321/1
+        :60F:C110522EUR3236,28
+        :61:1105240524D9,N192NONREF
+        :86:GIRO   428428 KPN - DIGITENNE    BETALINGSKENM.  000000042188659
+        5314606715                       BETREFT FACTUUR D.D. 20-05-2011
+        INCL. 1,44 BTW
+        :61:1105210523D11,59N426NONREF
+        :86:BEA   NR:XXX1234   21.05.11/12.54 DIRCKIII FIL2500 KATWIJK,PAS999
+        :61:1105230523D11,63N426NONREF
+        :86:BEA   NR:XXX1234   23.05.11/09.08 DIGROS FIL1015 KATWIJK Z,PAS999
+        :61:1105220523D11,8N426NONREF
+        :86:BEA   NR:XXX1234   22.05.11/14.25 MC DONALDS A44 LEIDEN,PAS999
+        :61:1105210523D13,45N426NONREF
+        :86:BEA   NR:XXX1234   21.05.11/12.09 PRINCE FIL. 55 KATWIJK Z,PAS999
+        :61:1105210523D15,49N426NONREF
+        :86:BEA   NR:XXX1234   21.05.11/12.55 DIRX FIL6017 KATWIJK ZH ,PAS999
+
+        :61:1105210523D107,N426NONREF
+        :86:BEA   NR:XXX1234   21.05.11/12.04 HANS ANDERS OPT./056 KAT,PAS999
+        :61:1105220523D141,48N426NONREF
+        :86:BEA   NR:XXX1234   22.05.11/13.45 MYCOM DEN HAAG  S-GRAVEN,PAS999
+        :62F:C110523EUR876,84
+      MT940
+      stub = instance_double(Cmxl::Statement)
+      allow(Cmxl::Statement).to receive(:new).and_return(stub)
+
+      Cmxl.parse(mt940_file('mt940-abnamro'))
+
+      expect(Cmxl::Statement).to have_received(:new).with(expected_data)
+    end
+
+    it 'splits the file into two statements' do
+      allow(Cmxl::Statement).to receive(:new)
+
+      Cmxl.parse(mt940_file('mt940-abnamro'))
+
+      expect(Cmxl::Statement).to have_received(:new).twice
+    end
+  end
+
+  describe 'MT940 handelsbank' do
+    it 'splits the file with the special characters correctly' do
+      expected_data = <<~MT940.chomp
+        채4:
```

Comment:
Note: here does is the file parsing quite strange. Yet the documentation is [also strange](https://www.handelsbanken.com/tron/xgpu/info/contents/v1/document/72-111414).